sigstore-java/src/main/java/dev/sigstore/tuf/Updater.java

```diff
@@ -78,24 +76,34 @@ public static Builder builder() {
   public void update()
       throws IOException, NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException {
     var root = updateRoot();
-    var timestamp = updateTimestamp(root);
-    var snapshot = updateSnapshot(root, timestamp);
-    updateTargets(root, snapshot);
+    // only returns a timestamp value if a more recent timestamp file has been found.
+    var timestampMaybe = updateTimestamp(root);
+    if (timestampMaybe.isPresent()) {
+      var snapshot = updateSnapshot(root, timestampMaybe.get());
+      updateTargets(root, snapshot);
+    }
   }
 
   // https://theupdateframework.github.io/specification/latest/#detailed-client-workflow
   Root updateRoot()
       throws IOException, RoleExpiredException, NoSuchAlgorithmException, InvalidKeySpecException,
-          InvalidKeyException, MetaFileExceedsMaxException, RollbackVersionException,
+          InvalidKeyException, FileExceedsMaxLengthException, RollbackVersionException,
           SignatureVerificationException {
     // 5.3.1) record the time at start and use for expiration checks consistently throughout the
     // update.
     updateStartTime = ZonedDateTime.now(clock);
 
     // 5.3.2) load the trust metadata file (root.json), get version of root.json and the role
     // signature threshold value
-
-    Root trustedRoot = GSON.get().fromJson(Files.readString(trustedRootPath), Root.class);
+    // TODO review from asraa@ do we need to verify the local root against our trusted root from the
```

Comment:
@asraa FYI

Reply:
Does "from the client" mean the embedded trust root? If so, no - because it may not be possible and the local root *should* be trusted (that being said, maybe there could be a fallback mechanism on the client root in case `updateRoots()` using the local trusted root doesn't pass)

Why it may not be possible: if the client embedded root is 1.root.json, and the local updated to version 3, then you've lost the 2.root.json that you needed in order to verify the chain. It would still exist on the remote and you could -- but then you're better off ALWAYS starting from the embedded root and always refreshing your cache. I'd say assume your cache is trusted

Reply:
Thanks for the detailed response! removing comment and leaving as is.

Reply:
Agree that's the best case for now.